component/src/main/wso2/kafka/consumer.bal

```diff
@@ -141,169 +139,169 @@ public type SimpleConsumer client object {
         return self.stop();
     }
 
-    public function __attach(service s, map<any> annotationData) returns error? {
-        return self.registerListener(s, annotationData);
+    public function __attach(service s, string? name) returns error? {
```

Comment:
`name` should be a defaultable param right? https://github.com/ballerina-platform/ballerina-lang/blob/master/stdlib/builtin/src/main/ballerina/builtin/listener.bal#L23
```suggestion
    public function __attach(service s, string? name = ()) returns error? {
```

This should ideally have failed at compile time, but doesn't at the moment due to a bug.